opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -0,0 +1,343 @@
+        self, *args, **kwargs
+    ) -> ObservableUpDownCounter:
+        with self._lock:
+            super().create_observable_up_down_counter(*args, **kwargs)
+            return self._meter.create_observable_up_down_counter(
+                *args, **kwargs
+            )
+
+
+class _DefaultMeter(Meter):
+    def create_counter(self, name, unit="", description="") -> Counter:
+        with self._lock:
+            super().create_counter(name, unit=unit, description=description)
+            return DefaultCounter(name, unit=unit, description=description)
+
+    def create_up_down_counter(
+        self, name, unit="", description=""
+    ) -> UpDownCounter:
+        with self._lock:
+            super().create_up_down_counter(
+                name, unit=unit, description=description
+            )
+            return DefaultUpDownCounter(
+                name, unit=unit, description=description
+            )
+
+    def create_observable_counter(
+        self, name, callback, unit="", description=""
+    ) -> ObservableCounter:
+        with self._lock:
+            return DefaultObservableCounter(
+                name,
+                callback,
+                unit=unit,
+                description=description,
+            )
+
+    def create_histogram(self, name, unit="", description="") -> Histogram:
+        with self._lock:
+            return DefaultHistogram(name, unit=unit, description=description)
+
+    def create_observable_gauge(
+        self, name, callback, unit="", description=""
+    ) -> ObservableGauge:
+        with self._lock:
+            return DefaultObservableGauge(
+                name,
+                callback,
+                unit=unit,
+                description=description,
+            )
+
+    def create_observable_up_down_counter(
+        self, name, callback, unit="", description=""
+    ) -> ObservableUpDownCounter:
+        with self._lock:
+            return DefaultObservableUpDownCounter(
+                name,
+                callback,
+                unit=unit,
+                description=description,
+            )
+
+
+_METER_PROVIDER = None
+_PROXY_METER_PROVIDER = None
+
+
+def get_meter(
+    name: str,
+    version: str = "",
+    meter_provider: Optional[MeterProvider] = None,
+) -> "Meter":
+    """Returns a `Meter` for use by the given instrumentation library.
+
+    This function is a convenience wrapper for
+    opentelemetry.trace.MeterProvider.get_meter.
+
+    If meter_provider is omitted the current configured one is used.
+    """
+    if meter_provider is None:
+        meter_provider = get_meter_provider()
+    return meter_provider.get_meter(name, version)
+
+
+def set_meter_provider(meter_provider: MeterProvider) -> None:
+    """Sets the current global :class:`~.MeterProvider` object.
+
+    This can only be done once, a warning will be logged if any furter attempt
+    is made.
+    """
+    global _METER_PROVIDER  # pylint: disable=global-statement
```

Comment:
should lock the global (missing in trace API I think)

Reply:
This should be done in a different PR because it also affects tracing.